src/librustc_metadata/encoder.rs

```diff
@@ -140,6 +143,40 @@ impl<'a, 'tcx> SpecializedEncoder<DefIndex> for EncodeContext<'a, 'tcx> {
     }
 }
 
+impl<'a, 'tcx> SpecializedEncoder<Span> for EncodeContext<'a, 'tcx> {
+    fn specialized_encode(&mut self, span: &Span) -> Result<(), Self::Error> {
+        if *span == DUMMY_SP {
+            return TAG_INVALID_SPAN.encode(self)
+        }
+
+        let span = span.data();
+
+        if span.lo > span.hi {
```

Comment:
As I mentioned in https://github.com/rust-lang/rust/issues/47066#issuecomment-355000084, this should not be possible when you use `Span`.
`Span` uses encoding `lo` + `len`, similarly to metadata, so it can't represent spans with `lo > hi`.